src/chainstate/stacks/block.rs

```diff
@@ -375,6 +387,13 @@ impl StacksMessageCodec for StacksBlock {
             ));
         }
 
+        if !StacksBlock::validate_transactions_static_epoch(&txs, epoch_id, true) {
+            warn!("Invalid block: target epoch is not activated");
+            return Err(codec_error::DeserializeError(
+                "Invalid block: target epoch is not activated".to_string(),
+            ));
+        }
```

Comment:
@jbencin @jcnelson removing this condition now, it is redundant because each tx is checked anyway in `read_next_at_most_with_epoch`